src/parser/SparqlParser.cpp

```diff
@@ -142,21 +142,11 @@ void SparqlParser::parseQuery(ParsedQuery* query, QueryType queryType) {
 
 // _____________________________________________________________________________
 void SparqlParser::parsePrologue(ParsedQuery* query) {
-  while (lexer_.accept("prefix")) {
-    lexer_.expect(SparqlToken::Type::IRI);
-    string key = lexer_.current().raw;
-    lexer_.expect(SparqlToken::Type::IRI);
-    string value = lexer_.current().raw;
-    addPrefix(key, value, query);
-  }
-}
-
-// _____________________________________________________________________________
-void SparqlParser::addPrefix(const string& key, const string& value,
-                             ParsedQuery* query) {
-  // Remove the trailing : from the key
-  SparqlPrefix p{key.substr(0, key.size() - 1), value};
-  query->_prefixes.emplace_back(p);
+  vector<SparqlPrefix> prefixes =
+      parseWithAntlr(sparqlParserHelpers::parsePrologue, *query);
+  query->_prefixes.insert(query->_prefixes.end(),
+                          std::make_move_iterator(prefixes.begin()),
+                          std::make_move_iterator(prefixes.end()));
```

Comment:
Yes the 'ql:' prefix is the one prefix that is already present in '_prefixes'. Moving it completely into a visit function might fail later if there is no prologue at all. I will add it to the default prefixes of the `_prefixMap` of the visitor.
On that note: is the default prefix "" -> "<>" in the Visitor needed? Looking at when it was changed it seems to be associated with the (wrong) baseDecl implementation.

Reply:
@ql-prefix: Yes, the default is fine for me. Maybe it is even cleaner to not put it there, but explicitly pass it to the constructor of the visitor (at the moment via `parseWithAntlr` which already has an argument for the (initial) prefix map of the visitor. Then this detail becomes more visible.

@empty default prefix: I am not sure, whether there is an empty prefix that is predefined. Consider it part of your scientific work to research that (try out other engines, e.g. the wikidata query service, and look into the standard).

Reply:
I had a quick look into the standards. If time allows it and it is reasonable I think that it should also be possible to implement the structure for correctly supporting the BASE IRI into the parser. So this would be the `PREFIX` directive and the implementation to process a present Base IRI. I think the biggest part will be to properly parser URLs into their components. Integrating it with the other components as per [RFC 3986 5.1.2/.3](https://datatracker.ietf.org/doc/html/rfc3986#section-5.1.2) might be out of scope for my work.